<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Louise A. Huuki-Myers">
<meta name="dcterms.date" content="2024-05-23">
<meta name="description" content="A high level overview of the spatialDLPFC project, now published in Science!">

<title>Louise A. Huuki-Myers - Spatial DLPFC: TL;DR</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../favicon-32x32.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Louise A. Huuki-Myers</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">About Me</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../blog.html"> 
<span class="menu-text">Blog</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../publications.html"> 
<span class="menu-text">My Publications</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/lahuuki/"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/louise-huuki/"> <i class="bi bi-linkedin" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/lahuuki/"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:lahuuki@gmail.com"> <i class="bi bi-envelope-at" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">Background</a>
  <ul class="collapse">
  <li><a href="#dlpfc" id="toc-dlpfc" class="nav-link" data-scroll-target="#dlpfc">DLPFC</a></li>
  <li><a href="#rna-sequencing" id="toc-rna-sequencing" class="nav-link" data-scroll-target="#rna-sequencing">RNA-sequencing&nbsp;</a>
  <ul class="collapse">
  <li><a href="#single-nucleus-rna-seq" id="toc-single-nucleus-rna-seq" class="nav-link" data-scroll-target="#single-nucleus-rna-seq">Single Nucleus RNA-seq</a></li>
  <li><a href="#spatially-resolved-transcriptomics-visium" id="toc-spatially-resolved-transcriptomics-visium" class="nav-link" data-scroll-target="#spatially-resolved-transcriptomics-visium">Spatially Resolved Transcriptomics (Visium)</a></li>
  <li><a href="#study-design" id="toc-study-design" class="nav-link" data-scroll-target="#study-design">Study Design&nbsp;</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#data-driven-spatial-domains" id="toc-data-driven-spatial-domains" class="nav-link" data-scroll-target="#data-driven-spatial-domains">Data-Driven Spatial Domains</a>
  <ul class="collapse">
  <li><a href="#spatial-registration-of-bayesspace-clusters" id="toc-spatial-registration-of-bayesspace-clusters" class="nav-link" data-scroll-target="#spatial-registration-of-bayesspace-clusters">Spatial Registration of <em>BayesSpace</em> Clusters</a></li>
  <li><a href="#novel-biology-in-spatial-domains" id="toc-novel-biology-in-spatial-domains" class="nav-link" data-scroll-target="#novel-biology-in-spatial-domains">Novel Biology in Spatial Domains</a></li>
  </ul></li>
  <li><a href="#single-nucleus-rna-seq-1" id="toc-single-nucleus-rna-seq-1" class="nav-link" data-scroll-target="#single-nucleus-rna-seq-1">Single Nucleus RNA-seq</a></li>
  <li><a href="#data-integration" id="toc-data-integration" class="nav-link" data-scroll-target="#data-integration">Data Integration</a>
  <ul class="collapse">
  <li><a href="#spot-deconvolution" id="toc-spot-deconvolution" class="nav-link" data-scroll-target="#spot-deconvolution">Spot Deconvolution</a></li>
  <li><a href="#spatially-map-disease-ligand-receptor-interactions" id="toc-spatially-map-disease-ligand-receptor-interactions" class="nav-link" data-scroll-target="#spatially-map-disease-ligand-receptor-interactions">Spatially Map Disease Ligand Receptor Interactions</a></li>
  </ul></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">Summary</a>
  <ul class="collapse">
  <li><a href="#data-availability" id="toc-data-availability" class="nav-link" data-scroll-target="#data-availability">Data Availability&nbsp;</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Spatial DLPFC: TL;DR</h1>
  <div class="quarto-categories">
    <div class="quarto-category">paper preview</div>
    <div class="quarto-category">spatialDLPFC</div>
    <div class="quarto-category">spatial</div>
    <div class="quarto-category">single cell</div>
  </div>
  </div>

<div>
  <div class="description">
    <p>A high level overview of the spatialDLPFC project, now published in <em>Science</em>!</p>
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Louise A. Huuki-Myers </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 23, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>This blog post provides a high-level summary of our paper <a href="https://doi.org/10.1126/science.adh1938">“A data-driven single cell and spatial transcriptomic map of the human prefrontal cortex”</a> published in <em>Science</em> in May 2024 (aka <strong>spatialDLPFC</strong>)<span class="citation" data-cites="Huuki-Myers2024">(<a href="#ref-Huuki-Myers2024" role="doc-biblioref">Huuki-Myers et al. 2024</a>)</span>.</p>
<p>In the spatialDLPFC project we set out to learn more about the organization of the dorsolateral prefrontal cortex (aka DLPFC), its cell types, and gene expression profile 🧠.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/SpatialDLPFC_GraphAbs_revision2.4.png" class="img-fluid figure-img"></p>
<figcaption>Graphical abstract for the spatialDLPFC project, summarized here (TODO add link)</figcaption>
</figure>
</div>
</section>
<section id="background" class="level1">
<h1>Background</h1>
<section id="dlpfc" class="level2">
<h2 class="anchored" data-anchor-id="dlpfc">DLPFC</h2>
<p>The dorsolateral prefrontal cortex region of the brain is especially important for executive functions including working memory, cognitive flexibility, and planning. Disruptions of the DLPFC have been associated with several psychiatric and neurodevelopmental disorders, including schizophrenia and autism spectrum disorder.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/background.png" class="img-fluid figure-img"></p>
<figcaption>Location of the DLPFC, its laminar structure (illistration from <span class="citation" data-cites="House_Pansky">(<a href="#ref-House_Pansky" role="doc-biblioref">House and Pansky, n.d.</a>)</span>), and major cell types.</figcaption>
</figure>
</div>
</section>
<section id="rna-sequencing" class="level2">
<h2 class="anchored" data-anchor-id="rna-sequencing">RNA-sequencing&nbsp;</h2>
<p>One of the ways that we can understand the functions of different cell types and structures in the brain is to study what genes they express by sequencing the RNA in a tissue. Recently, several advanced transcriptomic<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> approaches using RNA sequencing have emerged, enhancing our ability to analyze gene expression in the brain.</p>
<script defer="" class="speakerdeck-embed" data-slide="3" data-id="044430d9444f4ce1b9e016125f22cad2" data-ratio="1.7772511848341233" src="//speakerdeck.com/assets/embed.js"></script>
<p>This LEGO brain schematic demonstrates the evolution from bulk RNA sequencing, which provides a mixture of cell types, to single cell/single nucleus RNA-seq, which reveals the transcriptional profiles of individual cell types. The latest advancement, spatial transcriptomics, links gene expression to specific anatomical locations, providing deeper insights into the relationships between brain structure and function.</p>
<section id="single-nucleus-rna-seq" class="level3">
<h3 class="anchored" data-anchor-id="single-nucleus-rna-seq">Single Nucleus RNA-seq</h3>
<p>Single nucleus or single cell RNA sequencing (snRNA-seq) enables us to examine the gene expression of individual cells or nuclei. This technique relies on uniquely barcoded gel beads that attach to a single cell or nucleus, tagging all RNA molecules from that cell. When sequenced, these tagged RNA molecules can be traced back to their original cell. Cells or nuclei are then typically clustered by their gene expression profiles to identify different cell type populations. The expression profiles and cluster assignments are often visualized using reduced dimension plots such as UMAPs or tSNE. In these plots, each point represents a cell, and the distance between points indicates their similarity<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>; closer points represent more similar cells, which are often of the same cell type (shown by different colors).</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/background_sn.png" class="img-fluid figure-img"></p>
<figcaption>Cartoon of 10x snRNA-seq process (via 10x Genomics), and tSNE plot</figcaption>
</figure>
</div>
<p>In this experiment we are working with nuclei, as the cell membrane is destroyed when the brain tissue is frozen. The major cell type populations to identify in the DLPFC are neurons (Excitatory and Inhibitory), glial cells (ex: Astrocytes, Microglia, Oligodendrocytes, OPC), and vascular cells (Endothelial &amp; Mural).</p>
</section>
<section id="spatially-resolved-transcriptomics-visium" class="level3">
<h3 class="anchored" data-anchor-id="spatially-resolved-transcriptomics-visium">Spatially Resolved Transcriptomics (Visium)</h3>
<p>Spatially resolved transcriptomics maps RNA to specific locations on a tissue sample, allowing us to profile gene expression across anatomical features such as blood vessels, glands, or, in our case, layers of the brain’s cortex.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/background_spatial.png" class="img-fluid figure-img"></p>
<figcaption>Cartoon of Visium spatial transcriptomics (via 10x genomics), and example spot plots</figcaption>
</figure>
</div>
<p>We used Visium slides, which feature a grid of approximately 5,000 spots arranged in a 6.5x6.5 mm area. Each spot has a unique barcode that binds to the RNA in the contacted tissue. When the RNA is sequenced, these molecules can be traced back to their specific grid locations, similar to the barcodes in snRNA-seq.&nbsp;</p>
<p>This RNA-seq data is paired with a high-definition histology image of the original tissue, providing additional information and aiding in data visualization. We can visualize the gene expression of each spot in “spot plots” using color gradients overlaid on these images. In the example above we highlight the location of the gray matter with <em>SNAP25</em> a gene highly expressed in neurons, <em>MBP</em> highlights white matter, and <em>PCP4</em> marks layer 5.</p>
</section>
<section id="study-design" class="level3">
<h3 class="anchored" data-anchor-id="study-design">Study Design&nbsp;</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/study_design.png" class="img-fluid figure-img"></p>
<figcaption>Study design for spatialDLPFC</figcaption>
</figure>
</div>
<p>In this study we analyzed the DLPFC of ten healthy adult donors. We sampled three locations of the DLPFC: the anterior, middle, and posterior. All 30 samples were analyzed with Visium spatial transcriptomics, 19 (about 2 from each donor) were selected for snRNA-seq.</p>
</section>
</section>
</section>
<section id="data-driven-spatial-domains" class="level1">
<h1>Data-Driven Spatial Domains</h1>
<p>An earlier study, from the Lieber Institute, of spatial transcriptomics in the DLPFC <span class="citation" data-cites="maynard_2021">(<a href="#ref-maynard_2021" role="doc-biblioref">Maynard et al. 2021</a>)</span> relied on manually annotating the known layers of the cortex based on the histological images and the expression of select genes. This dataset has been invaluable for testing methodologies in spatial transcriptomics. However, manual annotation is tedious, time-consuming, and prone to human error and bias.&nbsp;</p>
<p>In our current study, which builds on the previous DLPFC project, we aimed to use unsupervised clustering to annotate the layers of the DLPFC, thereby avoiding the labor-intensive process of manual annotation and potentially discovering novel or unknown layers in the brain.</p>
<p>Based on benchmarking against the manually annotated layer data, we chose the method <em>BayesSpace</em> as the best method for clustering spatial data. We clustered the 30 Visium slides at a large range of resolutions, from k=2 to 28 (k denotes the number of clusters). We refer to these clusters as spatial domains, to name these domains we used the syntax <span class="math inline">\(Sp_{k}D_{d}\)</span>, where <em>k</em> is clustering resolution and <em>d</em> is spatial domain number, so <span class="math inline">\(Sp_{9}D_{1}\)</span> is spatial domain 1 when <em>k</em>=9.&nbsp;</p>
<p>We found that <em>k</em>=2 did a great job separating the white matter from the gray matter. With an increasing number of clusters, the layers of the cortex begin to emerge. This brings us to a question: which level of clustering best captures biologically important layers of the DLPFC?</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/clusters.png" class="img-fluid figure-img"></p>
<figcaption>A. Histological images of three DLPFC tissue sections B. spatial clustering at k=2, 9, and 16</figcaption>
</figure>
</div>
<section id="spatial-registration-of-bayesspace-clusters" class="level2">
<h2 class="anchored" data-anchor-id="spatial-registration-of-bayesspace-clusters">Spatial Registration of <em>BayesSpace</em> Clusters</h2>
<p>To check which resolution of <em>BayesSpace</em> clusters best matches the six histological layers plus white matter, we used a useful analysis we’ve developed called “spatial registration”. We will delve into the details of this analysis in a future blog post, and its application in this <a href="https://research.libd.org/spatialLIBD/articles/guide_to_spatial_registration.html">vignette</a>.</p>
<p>Briefly this analysis compares the gene expression profile of a reference set of clusters such as spatial regions or domains, annotated features, or cell type populations etc. (in this case the manual annotations from the pilot dataset), to a query set of clusters we want to learn more about (the <em>BayesSpace</em> clusters). The <em>t</em>-statistics from an enrichment analysis in the query and the reference set are correlated, pairwise across all groups. We visualize this in a heatmap where the high correlation is green, low correlation is purple.&nbsp; Where a query cluster has high correlation with a reference cluster, we can say the two groups are associated, and if the correlation passes our threshold we annotate the query group with the reference.&nbsp;</p>
<p>In the below example <span class="math inline">\(Sp_{7}D_{7}\)</span> has a high correlation with the manual annotation white matter, we then annotate it as&nbsp;<span class="math inline">\(Sp_{7}D_{7}\sim WM\)</span>. This annotation helps add biological context to our newly defined spatial domains.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/spatial_registration.jpeg" class="img-fluid figure-img"></p>
<figcaption>Example spatial registration between manual layers and <em>k</em>=7 <em>BayesSpace</em> clusters</figcaption>
</figure>
</div>
<p>From this process we learned that <em>k</em>=9 best reiterated the expected pattern of six layers + white matter, by matching each spatial domain to only one layer. In contrast to the <em>k</em>=7 resolution where some of the spatial domains (<span class="math inline">\(Sp_{7}D_{2}\)</span> and <span class="math inline">\(Sp_{7}D_{3}\)</span>) matched more than one layer. <em>K</em>=9 split white matter and Layer 1 into two spatial domains with unique gene expression.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/bayesSpace_k09_spatial_registration_heatmap_color.png" class="img-fluid figure-img"></p>
<figcaption><em>BayesSpace</em> <em>k</em>=9 cluster spatial registration vs.&nbsp;manual layers</figcaption>
</figure>
</div>
<p>For higher resolution clustering, <em>k</em>=16 was determined to be the optimal number of clusters based on the fast H+ statistic, so based on the data this is the best way to cluster the data. This further split the six original layers into 2-3 sub-layers each. The maximum number of clusters we could comfortably run on our computing setup was <em>k</em>=28, at this high number of clusters we lose the laminar definition.</p>
</section>
<section id="novel-biology-in-spatial-domains" class="level2">
<h2 class="anchored" data-anchor-id="novel-biology-in-spatial-domains">Novel Biology in Spatial Domains</h2>
<p>So what does all this clustering and layer matching help us learn about the brain?</p>
<p>At each resolution differentially expressed genes were detected between the spatial domains, this shows the complex organization of gene expression across the DLPFC tissue.&nbsp;</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Sp09D01_CLDN5.png" class="img-fluid figure-img"></p>
<figcaption>Clustering at <em>k</em>=9 highlighting <span class="math inline">\(Sp_{9}D_{1}\sim L1\)</span>, spot plots of and Violin plots of <em>CLDN5</em> expression</figcaption>
</figure>
</div>
<p>The data-driven clustering at <em>k</em>=9 revealed a sub-layer of the white matter with as much difference in gene expression that exists in the previously considered layers. It also found a thin band of vascular tissue (<span class="math inline">\(Sp_{9}D_{1}\sim L1\)</span>) in layer 1 with high expression for endothelial genes like <em>CLDN5</em>. These were both novel findings resulting from the unsupervised clustering. The sub-layers found in <em>k</em>=16 had distinct gene expression profiles.&nbsp;</p>
<p>These new spatial domains help refine the layered anatomy of the DLPFC. Neat! 🎉</p>
</section>
</section>
<section id="single-nucleus-rna-seq-1" class="level1">
<h1>Single Nucleus RNA-seq</h1>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Brain Cell Types Text.png" class="img-fluid figure-img"></p>
<figcaption>Cartoons of brain cell types, Created with BioRender.com</figcaption>
</figure>
</div>
<p>On the single nucleus side of the experiment, we processed 56k nuclei from n=19 samples. The first round of clustering (hierarchical clustering) found 29 distinct cell type clusters from seven broad cell types (note the abbreviations):&nbsp;</p>
<p><strong>Glia &amp; Vascular cells: provide structure to the brain, support neurons</strong><a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>&nbsp;</p>
<ol type="1">
<li><p>Astrocytes (Astro):link neurons to blood supply, clear neurotransmitters&nbsp;</p></li>
<li><p>Endothelial/Mural cells (EndoMural): blood vessels/vascular tissue</p></li>
<li><p>Microglia (Micro): immune function</p></li>
<li><p>Oligodendrocytes (Oligo): myelin sheath</p></li>
<li><p>Oligodendrocyte Precursor cells (OPCs)</p></li>
</ol>
<p><strong>Neurons: send and receive signals in the brain</strong></p>
<ol start="6" type="1">
<li><p>Excitatory Neurons (Excit)</p></li>
<li><p>Inhibitory Neurons (Inhib)<br>
</p></li>
</ol>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/TSNE_cellType.png" class="img-fluid figure-img"></p>
<figcaption>tSNE plot of snRNA-seq with 29 hierarchical clusters</figcaption>
</figure>
</div>
<p>Sub-populations in EndoMural, Oligos, and the Excit/Inhib Neurons were found in the first round of clustering.</p>
<p>In the DLPFC we know that different populations of excitatory neurons exist between the six layers of gray matter. To annotate our 13 Excit clusters we brought back our spatial registration tool, comparing all of the 29 hierarchical clusters to the manually annotated clusters from <span class="citation" data-cites="maynard_2021">(<a href="#ref-maynard_2021" role="doc-biblioref">Maynard et al. 2021</a>)</span> as well as the <em>BayesSpace</em> spatial domains at <em>k</em>=9 &amp; <em>k</em>=16.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/spatial_registration_sn_heatmap_bayesAnno.png" class="img-fluid figure-img"></p>
<figcaption>Spatial registration between the 29 snRNA-seq hierarchical clusters vs.&nbsp;histological layers or spatial domains at k=9 &amp; 16. Annotations with good confidence (cor &gt; 0.25, merge ratio = 0.1) are marked with “X” and poor confidence are marked with “*”.</figcaption>
</figure>
</div>
<p>We found Oligo and OPC cell types mapped to white matter, and EndoMural plus Astro mapped to Layer 1. Inhib neurons had a weak association with Layer 2-4, and the Excit neurons had strong associations with 1-3 layers each across the gray matter.&nbsp;The same patterns were found and re-fined in the spatial domains, such as the EndoMural groups mapping to <span class="math inline">\(Sp_{9}D_{1}\sim L1\)</span>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/TSNE_cellType_layer_label.png" class="img-fluid figure-img"></p>
<figcaption>tSNE plot of snRNA-seq data with layer level annotations</figcaption>
</figure>
</div>
<p>The layer associations were used to annotate the excitatory neuron populations by their strongest associated layers, other cell types were collapsed to their broad cell types. This resulted in our “layer-level” annotation with 13 cell types, and 7 populations of Excit neurons.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/markers_heatmap_layer_mean.png" class="img-fluid figure-img"></p>
<figcaption>Heatmap of the scaled pseudo-bulked logcounts for the top 10 marker genes for each layer level cell type</figcaption>
</figure>
</div>
<p>For each cell type we identified cell type specific marker genes with the <em>Mean Ratio</em> method described in <span class="citation" data-cites="huuki-myers">(<a href="#ref-huuki-myers" role="doc-biblioref">Huuki-Myers et al., n.d.</a>)</span>. The end product is gene expression profiles for layer annotated cell types in the human DLPFC! 🦠</p>
</section>
<section id="data-integration" class="level1">
<h1>Data Integration</h1>
<p>With this combined spatial and snRNA-seq data, there are a number of interesting downstream analyzes possible. Here I will briefly touch on two ways we integrated these data types.</p>
<section id="spot-deconvolution" class="level2">
<h2 class="anchored" data-anchor-id="spot-deconvolution">Spot Deconvolution</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/spot_deconvolution.jpeg" class="img-fluid figure-img"></p>
<figcaption>Overview of spot deconvolution: multiple cells exist in each spot, deconvolution predicts the cell type composition of each spot.</figcaption>
</figure>
</div>
<p>A challenge with Visium spatial transcriptomics is that each spot is larger than single cell resolution, and on average contains 3 cells per spot. To better understand the gene expression of each spot, we employed an analysis called spot deconvolution, which predicts what cell types exist in the tissue for each Visium spot.&nbsp;</p>
<script defer="" class="speakerdeck-embed" data-slide="39" data-id="ca75f869c0764642b48e8fa7143218ff" data-ratio="1.7772511848341233" src="//speakerdeck.com/assets/embed.js"></script>
<p>We determined that the methods <em>Tangram</em> and <em>Cell2location</em> were the most accurate for predicting cell type compositions through a benchmark experiment.&nbsp;From there we predicted the cell type composition of the spots across the 30 Visium slides with both deconvolution methods.</p>
<p>The spot deconvolution work was performed by <a href="https://www.linkedin.com/in/nick-eagles7/">Nick Eagles</a>. Check out his <a href="https://speakerdeck.com/nickeagles/libd-seminar-spot-deconvolution">spot deconvolution slide deck</a> above for more details.</p>
</section>
<section id="spatially-map-disease-ligand-receptor-interactions" class="level2">
<h2 class="anchored" data-anchor-id="spatially-map-disease-ligand-receptor-interactions">Spatially Map Disease Ligand Receptor Interactions</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/ligend_receptor_spatialDLPFC.jpeg" class="img-fluid figure-img"></p>
<figcaption>Cell-cell communication, <em>EFNA5</em> and <em>EPHA5</em> co-localizing in <span class="math inline">\(Sp_{9}D_{7}\sim L6\)</span> , cartoon of LR interaction in a Visium spot</figcaption>
</figure>
</div>
<p>To show how this dataset can be a rich resource to study neuropsychiatric diseases we explored the spatial location of a ligand-receptor (LR) interaction that is associated with schizophrenia. We performed a cell-cell communication analysis which predicts which cell types are interacting with each other, and then identified overlapping LR pairs with risk of schizophrenia from databases with the cell-cell communication results. From the common set of LR pairs we examined ligand <em>EFNA5</em> &amp; receptor <em>EPHA5</em>. From the snRNA-seq populations, <em>EFNA5</em> was most expressed in Excit_L5/6, and <em>EPHA5</em> in Excit_L6. From the Visium data we identified spots where the two genes were co-expressed, most frequent in <span class="math inline">\(Sp_{9}D_{7}\sim L6\)</span>, these spots also had high proportions of Excit_L5/6 neurons and Excit_L6 neurons predicted by spot deconvolution. Spatially mapping LR pairs helps us gain insight into the potentials for drug development. (This cool work was completed by <a href="https://x.com/BoyiGuo">Boyi Guo</a> and <a href="https://x.com/mgrantpeters">Melissa Grant-Peters</a>)</p>
<p>This analysis used many elements of the data from the spatialDLPFC project, and is just one example of how this dataset is relevant to the study of disease. In another application we also checked for enrichment of depression and PTSD related genes between the spatial domains. There are lots of exciting applications for the study of diseases with spatial and single cell, stay tuned to future work from the Lieber Institute for more! 👀</p>
</section>
</section>
<section id="summary" class="level1">
<h1>Summary</h1>
<p><img src="images/SpatialDLPFC_logo.png" class="img-fluid"></p>
<p>Overall we’ve created a paired spatial transcriptomic and single nucleus RNA-seq dataset of the human DLPFC. We’ve used spatial registration to map the new spatial domains and excitatory neurons to the classical histological layers. The data-driven spatial domains refine the layers of the DLPFC, finding laminar domains and cortical sub-layers. Spot deconvolution further refines the profile of each spot. This data has many applications in the study of neuropsychiatric diseases. We’ve made this dataset widely available to the scientific community (see below).</p>
<p>For more details be sure to check out our recently published paper in <em>Science</em> <span class="citation" data-cites="Huuki-Myers2024">(<a href="#ref-Huuki-Myers2024" role="doc-biblioref">Huuki-Myers et al. 2024</a>)</span> 🎉<a href="https://doi.org/10.1126/science.adh1938" class="uri">https://doi.org/10.1126/science.adh1938</a></p>
<section id="data-availability" class="level2">
<h2 class="anchored" data-anchor-id="data-availability">Data Availability&nbsp;</h2>
<p>The 30 DLPFC Visium samples &amp; the 56k nuclei snRNA-seq dataset are available to explore on our <a href="https://research.libd.org/spatialDLPFC/">interactive websites</a> and Bioconductor/R package <a href="https://research.libd.org/spatialLIBD/">spatialLIBD</a>.</p>
<p>Check out how your favorite gene is expressed over the layers or cell types of the DLPFC!</p>
<p><img src="images/SpatialDLPFC_shiny-01.png" class="img-fluid"></p>



</section>
</section>


<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-House_Pansky" class="csl-entry" role="listitem">
House, Earl Lawrence, and Ben Pansky. n.d. <em>A Functional Approach to Neuroanatomy</em>. 2nd ed. Blakiston Division.
</div>
<div id="ref-huuki-myers" class="csl-entry" role="listitem">
Huuki-Myers, Louise A., Kelsey D. Montgomery, Sang Ho Kwon, Sophia Cinquemani, Nicholas J. Eagles, Daianna Gonzalez-Padilla, Sean K. Maden, et al. n.d. <span>“Benchmark of Cellular Deconvolution Methods Using a Multi-Assay Reference Dataset from Postmortem Human Prefrontal Cortex.”</span> <a href="https://doi.org/10.1101/2024.02.09.579665">https://doi.org/10.1101/2024.02.09.579665</a>.
</div>
<div id="ref-Huuki-Myers2024" class="csl-entry" role="listitem">
Huuki-Myers, Louise A., Abby Spangler, Nicholas J. Eagles, Kelsey D. Montgomery, Sang Ho Kwon, Boyi Guo, Melissa Grant-Peters, et al. 2024. <span>“A Data-Driven Single-Cell and Spatial Transcriptomic Map of the Human Prefrontal Cortex.”</span> <em>Science</em>. <a href="https://doi.org/10.1126/science.adh1938">https://doi.org/10.1126/science.adh1938</a>.
</div>
<div id="ref-maynard_2021" class="csl-entry" role="listitem">
Maynard, KR, L Collado-Torres, LM Weber, C Uytingco, BK Barry, SR Williams, JL Catallini, et al. 2021. <span>“Transcriptome-Scale Spatial Gene Expression in the Human Dorsolateral Prefrontal Cortex.”</span> <em>Nature Neuroscience</em> 24 (3): 425–36. <a href="https://doi.org/10.1038/s41593-020-00787-0">https://doi.org/10.1038/s41593-020-00787-0</a>.
</div>
</div></section><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>the measurement of RNA transcription is known as “transcriptomics”<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>The full interpretation of these kinds of plots takes much nuance we won’t discuss here<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>The following are brief notes on cell type function to provide context, not comprehensive descriptions of the complex roles of these cell types<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/lahuuki\.github\.io\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://giscus.app/client.js" data-repo="lahuuki/lahuuki.github.io" data-repo-id="MDEwOlJlcG9zaXRvcnkzNDM1NTI0NjA=" data-category="General" data-category-id="DIC_kwDOFHoxzM4CfkDR" data-mapping="title" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="en" crossorigin="anonymous" async="">
</script>
<input type="hidden" id="giscus-base-theme" value="light">
<input type="hidden" id="giscus-alt-theme" value="dark">
</div> <!-- /content -->




</body></html>